test.js

```diff
@@ -0,0 +1,86 @@
+const ethers = require('ethers');
+const crypto = require('crypto');
+const axios = require('axios');
+
+//var id = crypto.randomBytes(32).toString('hex');
+//var privateKey = "0x"+id;
+//console.log("SAVE BUT DO NOT SHARE THIS:", privateKey);
+
+async function test() {
+	let quote, nonce, message, signature;
+
+	const wallet = new ethers.Wallet(process.env.TEST_PRIVATE_KEY);
+	console.log("Address: " + wallet.address);
+
+	// getQuote
+	await axios.post(`http://localhost:8081/getQuote`, {
+		type: "arweave",
+		userAddress: wallet.address,
+		files: [{length: 1256}, {length: 5969}],
+		payment: {
+			chainId: 80001,
+			tokenAddress: "0x0000000000000000000000000000000000001010",
+		},
+	})
+	.then((response) => {
+		quote = response.data;
+		console.log(`Got quote with id: ${quote.quoteId}`);
+	})
+	.catch((error) => {
+		console.error(error);
+		process.exit(1);
+	});
+
+	// upload
+	nonce = Math.floor(new Date().getTime()) / 1000;
+	message = ethers.utils.sha256(ethers.utils.toUtf8Bytes(quote.quoteId + nonce.toString()));
+	signature = await wallet.signMessage(message);
+
+	await axios.post(`http://localhost:8081/upload`, {
+		quoteId: quote.quoteId,
+		files: ["https://example.com/", "https://www.google.com/images/branding/googlelogo/1x/googlelogo_color_272x92dp.png"],
+		nonce: nonce,
+		signature: signature,
+	})
+	.then((response) => {
+		console.log("Upload worked");
+	})
+	.catch((error) => {
+		console.error(error);
+		process.exit(1);
+	})
+
+	// getStatus
+	for(let i = 0; i < 60; i++) {
+		let status
+		await axios.get(`http://localhost:8081/getStatus?quoteId=${quote.quoteId}`)
+		.then((response) => {
+			console.log(response.data);
+			status = response.data.status;
+		})
+		.catch((error) => {
+			console.error(error);
+		});
+		if(status == 5) break;
+		await new Promise(resolve => setTimeout(resolve, 1000));
+	}
+
+	// getLink
+	nonce = Math.floor(new Date().getTime()) / 1000;
+	message = ethers.utils.sha256(ethers.utils.toUtf8Bytes(quote.quoteId + nonce.toString()));
+	signature = await wallet.signMessage(message);
+
+	await axios.get(`http://localhost:8081/getLink?quoteId=${quote.quoteId}&nonce=${nonce}&signature=${signature}`)
```

Comment:
In a future iteration, I think it'd be worth refactoring these axios calls so the data is separate from the url, for readability.

```
axios.post('/user', {
    firstName: 'Fred',
    lastName: 'Flintstone'
  })
```

Reply:
I tried that, but since it was a GET, it didn't work. But, I am sure you are right, I just have to look up how to do it.

Reply:
https://axios-http.com/docs/api_intro
```
axios.get(url[, config])
```